aggregator/rollup_broadcaster.go

```diff
@@ -0,0 +1,150 @@
+package aggregator
+
+import (
+	"context"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
+	"github.com/Layr-Labs/eigensdk-go/chainio/txmgr"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/signerv2"
+	"github.com/ethereum/go-ethereum/common"
+
+	registryrollup "github.com/NethermindEth/near-sffl/contracts/bindings/SFFLRegistryRollup"
+	"github.com/NethermindEth/near-sffl/core/config"
+)
+
+const NUM_OF_RETRIES = 5
+const TX_RETRY_INTERVAL = time.Millisecond * 200
+
+type RollupWriter struct {
+	txMgr              txmgr.TxManager
+	client             eth.EthClient
+	sfflRegistryRollup *registryrollup.ContractSFFLRegistryRollup
+
+	logger logging.Logger
+}
+
+func NewRollupWriter(ctx context.Context, rollupInfo config.RollupInfo, signerConfig signerv2.Config, address common.Address, logger logging.Logger) (*RollupWriter, error) {
+	client, err := eth.NewClient(rollupInfo.RpcUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	chainId, err := client.ChainID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	signerV2, _, err := signerv2.SignerFromConfig(signerConfig, chainId)
+	if err != nil {
+		panic(err)
+	}
+	txMgr := txmgr.NewSimpleTxManager(client, logger, signerV2, address)
+
+	sfflRegistryRollup, err := registryrollup.NewContractSFFLRegistryRollup(rollupInfo.SFFLRegistryRollupAddr, client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RollupWriter{
+		txMgr:              txMgr,
+		client:             client,
+		sfflRegistryRollup: sfflRegistryRollup,
+		logger:             logger,
+	}, nil
+}
+
+func (w *RollupWriter) UpdateOperatorSet(ctx context.Context, message registryrollup.OperatorSetUpdateMessage, signatureInfo registryrollup.OperatorsSignatureInfo) error {
+	operation := func() error {
+		txOpts, err := w.txMgr.GetNoSendTxOpts()
+		if err != nil {
+			w.logger.Error("Error getting tx opts", "err", err)
+			return err
+		}
+
+		tx, err := w.sfflRegistryRollup.UpdateOperatorSet(txOpts, message, signatureInfo)
```

Comment:
I think we need to figure out what to do in case `id` that we want to send is larger than id that is in the contract.

The case basically when signatures for new message aggregated faster than for previous one.
I guess we need to add a queue to store such messages.

For now I added skipping in case of non-matching ids

Reply:
Ideally it should not be larger if the aggregator is also dealing with this - it's actually more likely that it is lower or equal. In this case I do think it's fine to just skip for now